lib/native_interaction/rubix/rubix_util.dart

```diff
@@ -263,6 +263,65 @@ class ExternalAccessTokenJWTClaim extends JwtClaim {
   }
 }
 
+class OrgAccessTokenJWTClaim extends JwtClaim {
+  OrgAccessTokenJWTClaim({
+    required String did,
+    required DateTime expiry,
+    required String peerId,
+    required String orgName,
+  }) : super(
+          issuer: _issuer,
+          subject: orgName,
+          maxAge: Duration(days: 30),
+          otherClaims: {
+            'type': _RubixTokenType.orgAccessToken.toString(),
+            'peerId': peerId,
+            'did': did,
+          },
+        );
+
+  factory OrgAccessTokenJWTClaim.fromJWTClaim(JwtClaim claim) {
+    return OrgAccessTokenJWTClaim(
+      orgName: claim.subject!,
+      expiry: claim.expiry!,
+      peerId: claim.toJson()['peerId'],
+      did: claim.toJson()['did'],
+    );
+  }
+
+  factory OrgAccessTokenJWTClaim.fromToken(String token) {
+    return OrgAccessTokenJWTClaim.fromJWTClaim(
+        verifyJwtHS256Signature(token, _secret));
+  }
+
+  String getOrgName() {
+    return subject!;
+  }
+
+  String getPeerId() {
+    return toJson()['peerId'];
+  }
+
+  String getDID() {
+    return toJson()['did'];
+  }
+}
+
+class OrgAccessToken {
+  static final int _accessTokenMaxAge = 30; // days
```

Comment:
token age might be something we have to worry about but not during a poc. We need to come back on this.